iceoryx2-ffi/ffi/src/api/sample_mut.rs

```diff
@@ -0,0 +1,271 @@
+#[no_mangle]
+pub unsafe extern "C" fn iox2_sample_mut_payload_mut(
+    sample_handle: iox2_sample_mut_ref_h,
+    payload_ptr: *mut *mut c_void,
+    payload_len: *mut c_size_t,
+) {
+    debug_assert!(!sample_handle.is_null());
+    debug_assert!(!payload_ptr.is_null());
+
+    let sample = &mut *sample_handle.as_type();
+
+    let payload = match sample.service_type {
+        iox2_service_type_e::IPC => sample.value.as_mut().ipc.payload_mut(),
+        iox2_service_type_e::LOCAL => sample.value.as_mut().local.payload_mut(),
+    };
+
+    *payload_ptr = payload.as_mut_ptr().cast();
+    if !payload_len.is_null() {
+        *payload_len = payload.len();
+    }
+}
+
+/// Acquires the samples payload.
+///
+/// # Safety
+///
+/// * `handle` obtained by [`iox2_publisher_loan()`](crate::iox2_publisher_loan())
+/// * `payload_ptr` a valid, non-null pointer pointing to a [`*const c_void`] pointer.
+/// * `payload_len` (optional) either a null poitner or a valid pointer pointing to a [`c_size_t`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_sample_mut_payload(
+    sample_handle: iox2_sample_mut_ref_h,
+    payload_ptr: *mut *const c_void,
+    payload_len: *mut c_size_t,
+) {
+    debug_assert!(!sample_handle.is_null());
+    debug_assert!(!payload_ptr.is_null());
+
+    let sample = &mut *sample_handle.as_type();
+
+    let payload = match sample.service_type {
+        iox2_service_type_e::IPC => sample.value.as_mut().ipc.payload(),
+        iox2_service_type_e::LOCAL => sample.value.as_mut().local.payload(),
+    };
+
+    *payload_ptr = payload.as_ptr().cast();
+    if !payload_len.is_null() {
+        *payload_len = payload.len();
+    }
+}
+
+/// Takes the ownership of the sample and sends it
+///
+/// # Safety
+///
+/// * `handle` obtained by [`iox2_publisher_loan()`](crate::iox2_publisher_loan())
+/// * `number_of_recipients`, can be null or must point to a valid [`c_size_t`] to store the number
+///                 of subscribers that received the sample
+#[no_mangle]
+pub unsafe extern "C" fn iox2_sample_mut_send(
+    sample_handle: iox2_sample_mut_h,
+    number_of_recipients: *mut c_size_t,
+) -> c_int {
+    debug_assert!(!sample_handle.is_null());
+
+    let sample_struct = &mut *sample_handle.as_type();
+    let service_type = sample_struct.service_type;
+
+    let sample = sample_struct
+        .value
+        .as_option_mut()
+        .take()
+        .unwrap_or_else(|| panic!("Trying to send an already sent sample!"));
+    (sample_struct.deleter)(sample_struct);
+
+    match service_type {
+        iox2_service_type_e::IPC => {
+            let sample = ManuallyDrop::into_inner(sample.ipc);
+            match sample.assume_init().send() {
+                Ok(v) => {
+                    if !number_of_recipients.is_null() {
+                        *number_of_recipients = v;
+                    }
+                }
+                Err(e) => {
+                    (sample_struct.deleter)(sample_struct);
```

Comment:
This needs to be removed. It leads to double deletion. I think I overlooked it when I fixed the mem leak.